File: 00-arrays-hashing/02-1-TwoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        m = {}
        for i, el in enumerate(nums): 
            rem = target - el
            if rem in m: 
                return [m[rem], i]
            else:
                m[el] = i
        return [-1, -1]

# print(Solution().twoSum([2,7,11,15], 9))

File: 00-arrays-hashing/02-1-TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let s = new Map();
  for (let i = 0; i < nums.length; i++) {
    let rem = target - nums[i];
    if (s.has(rem)) return [i, s.get(rem)];
    else s.set(nums[i], i);
  }
  return [-1, -1];
};
